az-digital/az_quickstart
# Fixes #2440 Publication Translations

Publications currently make use of the default language when sending information for citation processing, and the correct matching citation locale is not used. This PR adds locale and language support to some aspects of Publications.

Due to how the Drupal entity view system works, some fields were already translated correctly. Nested entity references were not. (e.g. attached media, author references).

## Description
This PR translated entity references properly via `getTranslationFromContext()` and adds a `az_publication.locale_metadata` service for translating the current Drupal language context to `citation-style-language/locales` context. This allows citations to have properly localization matching their language. This entails things like the correct locale-specific punctuation and separators being used for quotations, the correct names for months in date ranges, etc.

## Related issues
#2440 

## How to test

- enable `az_publication`
- create a publication. Include at least one author linked to a Person, and a document media for the publication. Select a date such that the month is displayed in the citation, e.g. `Jun-Jul 2023`
- verify the citation is still displayed correctly and that the author links to the correct person and the title links to the correct document.
- enable `az_multilingual`
- Add an additional language at `/admin/config/regional/language`
- Turn on translations for `Author` entities and `Publication` nodes at `/admin/config/regional/content-language`
- Translate the Person linked to by one of your authors.
- Translate your publication. Edit a component of the author's name in one translation. Provide a different document file for one of the translations.
- Place the language switcher block.
- Switch between languages on your publication. Verify that the translated author name is used in the correct language, that it links to the correct translated person, and that the file document for the translation is used on the correct language. Verify that localization is correct in the citation (correct quote marks for the language, correct month names, etc)

**Note**: citation localization is only supported for the languages listed in `citation-style-language/locales`.

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
